encord/utilities/benchmark_utilities.py

```diff
@@ -0,0 +1,210 @@
+from typing import Dict, List
+
+from shapely.geometry import Polygon, box
+
+from encord.client import EncordClientProject
+
+
+def get_project_labels_consensus(
+    baseline_project: EncordClientProject,
+    comparing_projects: List[EncordClientProject],
+    label_types: List[str],
+    threshold: float = 0.7,
+) -> Dict[str, Dict[str, float]]:
+    """
+    Calculate performance metrics of a label generating agent when compared with consensus reached by other agents.
+
+    Args:
+        baseline_project:
+            The project to be evaluated.
+        comparing_projects:
+            The list of projects where consensus is extracted.
+        label_types:
+            The list of feature node hashes indicating objects from baseline_project to be evaluated.
+        threshold:
+            Used in objects comparison. Two objects' instances are considered the same object if their Jaccard
+            similarity coefficient is greater or equal than the threshold; otherwise, they represent distinct objects.
+
+    Returns:
+        Consensus score (precision and recall) for objects from baseline_project indexed by their feature node hashes.
+
+    Raises:
+        NotImplementedError:
+            If objects using skeleton shape have their feature node hashes in label_types.
+        ValueError:
+            If a LabelRow of an EncordClientProject has an invalid format.
+    """
+    label_types = set(label_types)  # Use set so 'in' operation is O(1) in the average case instead of O(n)
+
+    # Obtain LabelRows shared between baseline_project and projects in comparing_projects (same data_hash)
+    data_hash_to_label_hash = dict()
+    for label_row in baseline_project.get_project().label_rows:
+        if label_row["label_status"] == "LABELLED":
+            lr = baseline_project.get_label_row(label_row["label_hash"], get_signed_url=False)
+            data_hash_to_label_hash[label_row["data_hash"]] = [lr]
+    for project in comparing_projects:
+        for label_row in project.get_project().label_rows:
+            if label_row["label_status"] == "LABELLED" and label_row["data_hash"] in data_hash_to_label_hash:
+                lr = project.get_label_row(label_row["label_hash"], get_signed_url=False)
+                data_hash_to_label_hash[label_row["data_hash"]].append(lr)
+
+    # Find consensus within LabelRows that share the same data_hash and compare it with the corresponding LabelRow in
+    # baseline_project in order to extract consensus score
+    consensus_score = dict()
+    for label_row_list in data_hash_to_label_hash.values():
+        if len(label_row_list) == 1:  # there is no consensus data to compare with
+            continue
+
+        # Extract frames within the LabelRows and reorder them in the fashion:
```

Comment:
Nothing that I will block this review on, but usually if I need to add a comment like this I'd rather create a little helper function with a similar name to this comment. That way it might be also easier to read what is going on here.

Reply:
Also, if you create a nice helper function you might be able to type the response types better and someone else could also re-use your helper.